cmd/thanos/sidecar.go

```diff
@@ -269,13 +275,15 @@ func runSidecar(
 			return errors.Wrap(err, "setup gRPC server")
 		}
 
-		s := grpcserver.New(logger, reg, tracer, comp, promStore,
+		grpcHealthSrv := health.NewServer()
+		s := grpcserver.New(logger, reg, tracer, comp, grpcHealthSrv, promStore,
 			grpcserver.WithListen(grpcBindAddr),
 			grpcserver.WithGracePeriod(grpcGracePeriod),
 			grpcserver.WithTLSConfig(tlsCfg),
 		)
 		g.Add(func() error {
 			statusProber.Ready()
+			grpcHealthSrv.Resume()
```

Comment:
It's already difficult to not forget about proper statusProber management, not we have two probes to be instrumented. 

What do you think about:
* Wrapping health server as readiness prober
* Adding joinedProber which will regsiter both http Probes and our gRPC readiness probe (named health, but I assumed it will have rather readiness logic)
* Create joined Prober with gRPC + http for places with gRPC traffic as `Ready`, `NotReady` methods are already correctly instrumented there

WDYT?

Reply:
I really like the idea 👍 I'll work on it.